Omero CLI, Refactor R-omero installation

manics opened the pull request:
Install the OMERO CLI including support for imports. `omero` is in the default path so can be run without a prefix, e.g.
```
touch image.fake
omero login -s eel.openmicroscopy.org -u user-1
omero import image.fake
```
<img width="1121" alt="screen shot 2018-07-20 at 13 31 34" src="https://user-images.githubusercontent.com/1644105/43002604-4737be1c-8c21-11e8-944f-738c758f822c.png">
@pwalczysko 

- [x] --depends-on https://github.com/ome/rOMERO-gateway/pull/52

Related to https://trello.com/c/09QarUT5/37-basel-training-03-04-09-2018-setup

----
## Updated 2018-08-25

Additional changes:
- CellProfiler uses a conda JRE instead of a global one
- R is installed using a conda environment file
- OMERO Python2 Jupyter kernel logos have been changed from the OME logo to the OMERO python logo from https://www.openmicroscopy.org/omero/downloads/ (it'd be nice if we had an R one too!):
<img width="366" alt="screen shot 2018-07-26 at 10 41 31" src="https://user-images.githubusercontent.com/1644105/43255016-97ae1320-90c0-11e8-9173-f0eeeaacf015.png">

manics committed:
Move copy notebook to end to reduce cache churn

manics committed:
Include OMERO.server and install /usr/local/bin/omero wrapper

manics committed:
install.R is currently broken, use devel version

jburel commented:
Maybe open a PR with only ac08c11 so we can merge the fix 
then we can review the R changes

manics commented:
See https://github.com/ome/training-notebooks/pull/19

manics committed:
Test docker build

jburel commented:
rgateway tagged to 0.4.1

manics committed:
CellProfiler uses jvm in conda env instead of global

manics committed:
Replace Dockerfile r-omero prerequisites with a conda env file

manics committed:
Install OMERO using omego

manics committed:
Replace python2 jupyter OME logo with OMERO.py logo

manics committed:
COPY --chown is now supported by Docker Hub

jburel commented:
ac08c11dca4ba371ecb5a24e2bb8e614dc25ecb2 has been merged
so it can be removed from this PR

jburel commented on Dockerfile:
```diff
@@ -4,98 +4,52 @@ MAINTAINER ome-devel@lists.openmicroscopy.org.uk
 ADD docker/environment-python2-cellprofiler.yml .setup/
 RUN conda env update -n python2 -f .setup/environment-python2-cellprofiler.yml
+# CellProfiler has to be installed in a separate step because it requires
+# the JAVA_HOME environment variable set in the updated environment
```
Having that step of the requirement file implies that now that we will have to copy it in each Docker file we want to have it installed.
Docker-cp, Docker-all
Maybe there is no way around it

manics replied:
As discussed, although it's two steps it's balanced out by the overall reduction in complexity of the Dockerfile

jburel commented on docker/environment-r-omero.yml:
```diff
@@ -0,0 +1,24 @@
+dependencies:
+- bioconductor-ebimage=4.13.*
+- maven=3.5.*
+- openjdk=8.0.*
```
openjdk is in both requirement files.
I don't know if it is possible to combine requirements files
saying that the apps might not have the same required version over time

manics replied:
As discussed, it's easier to keep them separate.

jburel commented:
R and CP notebooks work as expected
Trying to import from terminal
```
omero import image.fake
Using session c2608e57-b783-4514-b4e0-42c4971ae2d5 (trainer-1@outreach.openmicroscopy.org:4064). Idle timeout: 10 min. Current group: Lab1
Traceback (most recent call last):
  File "/opt/omero/OMERO.server/bin/omero", line 125, in <module>
    rv = omero.cli.argv()
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1620, in argv
    cli.invoke(args[1:])
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1095, in invoke
    stop = self.onecmd(line, previous_args)
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1172, in onecmd
    self.execute(line, previous_args)
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1254, in execute
    args.func(args)
  File "/opt/omero/OMERO.server/lib/python/omero/plugins/import.py", line 494, in importer
    self.do_import(command_args, xargs)
  File "/opt/omero/OMERO.server/lib/python/omero/plugins/import.py", line 505, in do_import
    stdout=out, stderr=err)
  File "/opt/omero/OMERO.server/lib/python/omero/java.py", line 140, in popen
    check_java(command)
  File "/opt/omero/OMERO.server/lib/python/omero/java.py", line 31, in check_java
    raise Exception("Java could not be found. (Executable=%s)" % command[0])
Exception: Java could not be found. (Executable=java)
jovyan@f9355320be5c:~$

```

mtbc commented on Dockerfile:
```diff
@@ -4,98 +4,52 @@ MAINTAINER ome-devel@lists.openmicroscopy.org.uk
+
+# OMERO full CLI
+# This currently uses the python2 environment, should we move it to its own?
+ARG OMERO_VERSION=5.4.6
```
I guess we can use 5.4.7 imminently. :smiley:

manics replied:
Done!

dominikl commented:
Same here, notebooks (including R notebook) load fine, but same Java error in command line for 'omero' CLI.

manics committed:
omero cli needs to be in activated python2 conda env

manics committed:
Bump OMERO CLI to 5.4.7

manics commented:
Sorry about that, I forgot to re-test OMERO CLI after the JDK work for CellProfiler and R. Should be working now:
![screen shot 2018-07-26 at 10 26 01](https://user-images.githubusercontent.com/1644105/43254208-cadd049c-90be-11e8-9c54-c9f19dee8f68.png)

I've also rebased to include the Travis fix, and I've updated the CLI to 5.4.7, the python-omero conda package is still at 5.4.6 though: https://github.com/ome/training-notebooks/blob/35ce8f1d13dc03e2d729e7550095cd23d9d3e555/docker/environment-python2-omero.yml#L6

manics committed:
Make pip/conda/omego less verbose to reduce log noise

Note install.R is still very verbose

manics committed:
Update to rOMERO 0.4.2, use --quiet flag

manics commented:
Updated to rOMERO 0.4.2

dominikl commented:
Looks good, rOMERO 0.4.2 installed, R and Cellprofiler notebook can be loaded and the CLI import mentioned in the description works too! 👍

manics committed:
Pin approx version for CellProfiler dependencies

jburel commented:
With ffc8484 we can now rebuild the image

manics committed:
Bump python-omero to 5.4.7

manics commented:
The final commit updates the python-omero bioconda to 5.4.7
